dotCMS/src/main/java/com/dotcms/content/elasticsearch/util/RestHighLevelClientProvider.java

```diff
@@ -0,0 +1,40 @@
+package com.dotcms.content.elasticsearch.util;
+
+import com.dotmarketing.util.Config;
+import com.dotmarketing.util.Logger;
+import org.elasticsearch.client.RestHighLevelClient;
+
+/**
+ * Provider used to load custom High Level Rest clients to handle API requests in Elastic
+ * By default, {@link DotRestHighLevelClientProvider} is used if `ES_REST_CLIENT_PROVIDER_CLASS` is not set
+ * @author nollymar
+ */
+public abstract class RestHighLevelClientProvider {
+
+    private static RestHighLevelClientProvider INSTANCE;
+
+    public static RestHighLevelClientProvider getInstance() {
+        if (INSTANCE == null) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Singleton is not thread safe](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=3712629)